### ISSUE TITLE
CORS decorator

### ISSUE THREAD

**Issue by dmzaytsev** [yegor256/takes#258]
At the moment Takes doesn't support [CORS](http://www.w3.org/TR/cors/) 
I believe a CORS decorator has to be. It allows browser's request to Takes-based resources on another domain.

I think we can start from this
```
class TkCors implements Take {
  TkCors(Take take, String ... domains)
}
```
* If the `Origin` header not present in the request skip all steps below.
* If the `Origin` header is present in the request and the domains list is empty add a single `Access-Control-Allow-Origin` header to the response with  the value of the `Origin` header.
* If in previous step the domains list is not empty then the `Origin` header should match for any of the values in list of domains. Add same header as step above. If header doesn't match skip all steps. 
* Add a single `Access-Control-Allow-Credentials` header with the string `true` as value.
* Add a single  `Access-Control-Allow-Methods` header with the string `OPTIONS,GET,PUT,POST,DELETE,HEAD` as value.

**Comment by yegor256** [comment]
@dmzaytsev good idea, thanks... I would call it `TkCORS` though

**Comment by davvd** [comment]
@dmzaytsev we will find someone to do this task, soon

**Comment by davvd** [comment]
@dmzaytsev I set milestone here to `1.0`, let me know if it is wrong

**Comment by davvd** [comment]
@zshamrock please pick this up, and keep in mind [these instructions](http://www.xdsd.org/2014/04/17/how-xdsd-is-different.html). Any technical questions - ask right here... Task's budget is **30 mins** (see [this](http://www.xdsd.org/2014/04/17/how-xdsd-is-different.html) for explanation)

**Comment by davvd** [comment]
@dmzaytsev many thanks for the bug report, I added 15 mins to your acc, transaction `56932653`

**Comment by davvd** [comment]
@zshamrock you're holding this ticket for 15 days already. If it is not finished (and closed) in the next 48 hours, it will be re-assigned to someone else, see [No Obligations](http://www.xdsd.org/2014/04/13/no-obligations-principle.html) principle ([this article](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html) should help if you're stuck)... added -30 to your rating, now it is equal to [-30](http://www.netbout.com/b/35316?open=rating)

**Comment by davvd** [comment]
@zshamrock this task is taking too long, I have to change the performer, sorry. Please stop working with it right now. See our [no obligations](http://www.xdsd.org/2014/04/13/no-obligations-principle.html) principle... added -60 to your rating, now it is equal to [-120](http://www.netbout.com/b/35316?open=rating)

**Comment by yegor256** [comment]
@davvd I added the tag

**Comment by davvd** [comment]
@antonini this task is for you, keep [these principles](http://www.xdsd.org/2014/04/17/how-xdsd-is-different.html) in mind, and don't hesitate to ask any technical questions; This task's budget is **30 mins**. This is exactly how much will be paid when the problem explained above is solved. See [this](http://www.xdsd.org/2014/04/17/how-xdsd-is-different.html) for more information

**Comment by antonini** [comment]
@davvd PR #326 is available and waiting for review.

**Comment by davvd** [comment]
@dmzaytsev there is a puzzle in this code `258-12a59dfa`/#342, we'll resolve it later

**Comment by davvd** [comment]
@antonini you're working with this issue for 15 days already; -30 added to your rating, current score is: [+76](http://www.netbout.com/b/35346?open=rating)

**Comment by dmzaytsev** [comment]
@antonini thanks!

**Comment by antonini** [comment]
@davvd This issue is still as pending on my agenda.

**Comment by davvd** [comment]
@ypshenychka review this ticket please, for compliance with our [quality rules](http://www.teamed.io/qa.html)

**Comment by ypshenychka** [comment]
@davvd Quality is good.

**Comment by davvd** [comment]
> @davvd Quality is good.

@ypshenychka thanks for the review

**Comment by davvd** [comment]
@antonini 10 mins was added to the account of @ypshenychka (for QA review), in transaction 59118657

Much appreciated! **30 mins** was added to your account, payment ID `AP-7LX03090UJ9369809`, time spent is 361 hours and 42 mins

+30 added to your rating, current score is: [+136](http://www.netbout.com/b/35346?open=rating)

**Comment by davvd** [comment]
@dmzaytsev last puzzled solved `258-12a59dfa`/#342
